### PR TITLE
Add description of some integrated gunmods

### LINKED ISSUE #17816: Integrated mods are described in item info, but as "SEEING THIS IS A BUG"
* Debug in crafting skills
* Unlock recipes
* Look at entry for pipe crafting gun

### LINKED ISSUE #18029: LeMat revolver (if you're seeing this - it's a bug message)
#5382
Picrelated 
lemat

### PR DESCRIPTION
**Pull request by cainiaowu**
Fixed #18029.
Fixed #17816.

### CODE REVIEW

File: data/json/items/gunmod/underbarrel.json [hunk 1]
```diff
@@ -168,7 +168,7 @@
     "id": "ksg_aux_shotgun",
     "type": "GUNMOD",
     "name": "KSG second magazine",
-    "description": "IF YOU ARE SEEING THIS IT IS A BUG.",
+    "description": "The integrated sencond shotgun magazine of Kel-Tec KSG.",
```
Comment (woutershep) [inline data/json/items/gunmod/underbarrel.json]: typo of 'second'?

File: data/json/items/gunmod/underbarrel.json [hunk 2]
```diff
@@ -65,7 +65,7 @@
     "id": "combination_gun_shotgun",
     "type": "GUNMOD",
     "name": "combination gun shotgun",
-    "description": "IF YOU ARE SEEING THIS IT IS A BUG.",
+    "description": "The integrated underbarrel shotgun of combination gun.",
```
Comment (Izicata) [inline data/json/items/gunmod/underbarrel.json]: For generic home-made combination guns, there should be an "a" after the word "of", e.g. "The integrated underbarrel shotgun of a combination gun."
For specific makes of guns, there should be a "the" after the word "of", e.g. "The integrated second shotgun magazine of the Kel-Tec KSG".

**Comment by mugling** [comment]
It might be better to just not display info for `IRREMOVABLE` contents?

**Comment by cainiaowu** [comment]
Why not? More info to player never hurts.

**Comment by mugling** [comment]
> Why not?

Needless work keeping it updated

> More info to player never hurts.

What extra information is being provided here?

**Comment by Zireael07** [comment]
> What extra information is being provided here?

Gunmod descriptions.

**Comment by mugling** [comment]
> > What extra information is being provided here?
> 
> Gunmod descriptions.

To rephrase, what additional information is conveyed?

**Comment by BevapDin** [comment]
> It might be better to just not display info for  IRREMOVABLE  contents?

The description (and the name) is the *only* thing about those mods (the irremovable ones) that the user ever sees.

Looking at a LeMat revolver, I see it has an integrated mod "LeMat revolver shotgun", located in the "underbarrel" slot and that's it. What the mod does, what it is, why it's there etc. - no information at all.

> To rephrase, what additional information is conveyed?

The proposed description should be a bit more detailed, yes. Alternatively the item info window could show the properties of the contained mods as well and the mods description would not be need. The window is already scrollable, so it has enough space.

@cainiaowu Considering that the mods will only appear attached to a specific gun, it's not needed to repeat what gun that is, the user already knows that from the item info window. Interesting would be how many cartridges it can hold.

File: data/json/items/gunmod/underbarrel.json [hunk 3]
```diff
@@ -65,7 +65,7 @@
     "id": "combination_gun_shotgun",
     "type": "GUNMOD",
     "name": "combination gun shotgun",
-    "description": "IF YOU ARE SEEING THIS IT IS A BUG.",
+    "description": "The integrated underbarrel shotgun of a combination gun which holds a single shot.  It's irremoveable.",
```
Comment (Rivet-the-Zombie) [inline data/json/items/gunmod/underbarrel.json]: irremoveable -> irremovable

**Pull request merged**